api.js

```diff
@@ -1788,4 +1788,28 @@ API.prototype.setErrorGroupCallback = function setErrorGroupCallback(callback) {
   this.agent.errors.errorGroupCallback = callback
 }
 
+/**
+ * Function for assigning metadata to all LLM events. This should only
+ * be called once in your application and before executing any openai, bedrock,
+ * langchain, generative AI methods.
+ *
+ * If passing in metadata via `api.recordLlmFeedbackEvent`, it will take precedence
+ * over what is assigned wit this method.
+ *
+ * @param {object} metadata key/value metadata to be assigned to all LLM Events on creation
+ * @example
+ * newrelic.setLlmMetadata({ type: 'openai', framework: 'express', region: 'us' })
+ */
+API.prototype.setLlmMetadata = function setLlmMetadata(metadata) {
+  const metric = this.agent.metrics.getOrCreateMetric(NAMES.SUPPORTABILITY.API + '/setLlmMetadata')
+  metric.incrementCallCount()
+
+  if (!(metadata?.constructor === Object)) {
```

Comment:
I'm likely coming up with problems that don't exist, but the `constructor` property is writable. Someone _could_ change it before passing in their metadata object. What I think is _way_ more likely, though, is that someone creates an object from `null` and tries to pass it in. I think it would be more complete to resolve https://github.com/newrelic/node-newrelic/issues/1864 and use the `isSimpleObject` function.